alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -494,20 +377,27 @@ void RunningPlan::setFailedChild(const AbstractPlan* child)
     }
 }
 
+void RunningPlan::setFailureHandlingNeeded(bool failHandlingNeeded)
+{
+    if (failHandlingNeeded) {
+        _status.status = PlanStatus::Failed;
```

Comment:
why set status to failed immediately?

Reply:
This is not new code. The logic says that if a failure has to be handled, the plan status should signal failure. Reworking failure semantics is done in branch hs_rework_failure_handling.